sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -450,9 +451,19 @@ export class MessageSender extends LinkEntity {
           // It is possible for someone to close the sender and then start it again.
           // Thus make sure that the sender is present in the client cache.
           if (!this._sender) this._context.sender = this;
-          await this._ensureTokenRenewal();
+          this._ensureTokenRenewal();
         }
       } catch (err) {
+        if (this._tokenRenewalTimer) {
+          clearTimeout(this._tokenRenewalTimer);
+          this._tokenRenewalTimer = undefined;
+        }
+
+        if (this._sender) {
+          await this._sender.close();
+          this._sender = undefined;
+        }
```

Comment:
What made us add the above in the catch block?. The catch block is for when link initialization fails. The lock renewal is not set up until we have a successful link initialization. Neither is `this._sender` set until a successful link initialization. What am I missing?

Reply:
Nothing, I was trying to see if I could isolate what I'd missed that was causing the test failures - this is pretty much the only area related to an "eternal timer" that I touched. I can revert this particular change as I think it was all related to some potential hackery I was doing in our unit tests with setTimeout/clearTimeout (that I've removed).

Reply:
Removed. :)